account_financial_report/report/abstract_report_xlsx.py

```diff
@@ -531,7 +531,10 @@ def _get_currency_amt_format(self, line_object, report_data):
             format_amt = report_data["formats"]["format_amount"]
             field_prefix = "format_amount"
         if "currency_id" in line_object and line_object.get("currency_id", False):
-            currency = line_object["currency_id"]
+            if isinstance(line_object["currency_id"], int):
+                currency = self.env["res.currency"].browse(line_object["currency_id"])
+            else:
+                currency = line_object["currency_id"]
```

Comment:
I still don't get: if you have renamed the variable to `currency`, why are you loading here `line_object["currency_id"]`?

I also still see the duality `currency_id` / `currency`, and this is not the ideal, as we can't assure if both are available.

Reply:
I don't agree with making all these changes either.

Another possible approach (at template level only) could be:

Modify all records that use `currency_id` (https://github.com/OCA/account-financial-reporting/blob/14.0/account_financial_report/report/templates/general_ledger.xml#L324 for example) to something "_similar_" to https://github.com/OCA/account-financial-reporting/blob/14.0/account_financial_report/report/templates/general_ledger.xml#L586-L605 (define `account_currency` object and use it to display the value).

Reply:
You don't have to keep compatibility with both `currency_id` and `currency`. Just make it homogeneous across entire module, which was not previously.